### PR TITLE
Tools fixes part 2 - loggers + trie checker proper fix

### PR DESCRIPTION
**Pull request by iulianpascalau**
- trie checker proper fix
- updated to the newest logger version

### CODE REVIEW

**Review by sstanculeanu — APPROVED** [review]

File: trieTools/trieChecker/logParser/formatter.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package logParser
+
+import (
+	"strings"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+// TODO remove this package
```
Comment (andreibancioiu) [inline trieTools/trieChecker/logParser/formatter.go:10]: :rocket:

File: trieTools/trieChecker/main.go [hunk 2]
```diff
@@ -98,6 +101,13 @@ func checkTrie(flags trieToolsCommon.ContextFlagsConfig, mainRootHash []byte) er
 		log.LogIfError(errNotCritical)
 	}()
 
+	// TODO remove this workaround when the GetAllLeavesOnChannel gets refactored
+	formatter := logParser.NewLoggerFormatter()
+	err = logger.AddLogObserver(ioutil.Discard, formatter)
```
Comment (andreibancioiu) [inline trieTools/trieChecker/main.go:106]: It seems that `ioutil` is now deprecated (in favor of `io`).

File: trieTools/trieChecker/logParser/formatter.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// Output will always return nil byte slice but records any type of errors encountered
+func (formatter *loggerFormatter) Output(line logger.LogLineHandler) []byte {
+	if line.GetLogLevel() != int32(logger.LogError) {
+		return nil
+	}
+
+	data := append([]string{line.GetMessage()}, line.GetArgs()...)
+
+	formatter.mutErrors.Lock()
+	formatter.errorStrings = append(formatter.errorStrings, strings.Join(data, " "))
```
Comment (andreibancioiu) [inline trieTools/trieChecker/logParser/formatter.go:33]: Nice error interceptor :)

**Review by andreibancioiu — APPROVED** [review]

File: trieTools/trieChecker/logParser/formatter.go (hunk 3 above)
Comment (iulianpascalau) [reply trieTools/trieChecker/logParser/formatter.go:33]: it's a terrible hack, but the only one possible, for now :(

**Pull request merged**